include/circt/Dialect/Arc/ArcOps.td

```diff
@@ -566,6 +566,68 @@ def StateWriteOp : ArcOp<"state_write", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Simulation Orchestration
+//===----------------------------------------------------------------------===//
+
+def SimInstantiate : ArcOp<"sim.instantiate"> {
+  let summary = "Instantiates an Arc model for simulation";
+  let description = [{
+    Creates an instance of an Arc model in scope, in order to simulate it.
+  }];
+  let results = (outs SimModelInstance:$instance);
+  let assemblyFormat = [{ attr-dict `:` qualified(type($instance)) }];
+}
+
+def SimSetInput : ArcOp<"sim.set_input"> {
+  let summary = "Sets the value of an input of the model instance";
+  let description = [{
+    Sets the value of an input port in a specific instance of a model. The
+    provided input port must be of input type and contain an integer of the
+    right size.
+  }];
+  let arguments = (ins SimModelInstance:$instance,
+                       StrAttr:$input,
+                       AnyInteger:$value);
+  let assemblyFormat =
+    [{ $instance `,` $input `=` $value attr-dict
+       `:` type($value) `,` qualified(type($instance)) }];
+}
+
+def SimGetPort : ArcOp<"sim.get_port"> {
```

Comment:
This can probably have the `Pure` trait:
```suggestion
def SimGetPort : ArcOp<"sim.get_port", [Pure]> {
```

Reply:
I'd say it should have a read side-effect. If it is pure, it would be possible to move it over a `arc.sim.step` which can change the value it returns.

Reply:
Oh yeah good point. Keep forgetting about the memory ordering.